sycl/unittests/CMakeLists.txt

```diff
@@ -1,26 +1,38 @@
 add_custom_target(SYCLUnitTests)
 set_target_properties(SYCLUnitTests PROPERTIES FOLDER "SYCL tests")
 
-# add_sycl_unittest(test_dirname file1.cpp, file2.cpp ...)
+# add_sycl_unittest(test_dirname SHARED|OBJECT file1.cpp, file2.cpp ...)
 #
 # Will compile the list of files together and link against SYCL.
 # Produces a binary names `basename(test_dirname)`.
-function(add_sycl_unittest test_dirname)
+function(add_sycl_unittest test_dirname link_variant)
   # Enable exception handling for these unit tests
   set(LLVM_REQUIRES_EH 1)
 
   if (MSVC AND CMAKE_BUILD_TYPE MATCHES "Debug")
-    set(sycl_lib "sycld")
+    set(sycl_obj_target "sycld_object")
+    set(sycl_so_target "sycld")
   else()
-    set(sycl_lib "sycl")
+    set(sycl_obj_target "sycl_object")
+    set(sycl_so_target "sycl")
   endif()
 
-  add_unittest(SYCLUnitTests ${test_dirname} ${ARGN})
+  if ("${link_variant}" MATCHES "SHARED")
+    set(SYCL_LINK_LIBS ${sycl_so_target})
+    add_unittest(SYCLUnitTests ${test_dirname} ${ARGN})
+  else() 
+    add_unittest(SYCLUnitTests ${test_dirname} 
+                $<TARGET_OBJECTS:${sycl_obj_target}> ${ARGN})
```

Comment:
This line expands to list of all sycl's .o objects?

Reply:
Correct